module/advancement/types/item-grant.mjs

```diff
@@ -71,19 +71,17 @@ export class ItemGrantAdvancement extends Advancement {
     for ( const [uuid, selected] of Object.entries(data) ) {
       if ( !selected ) continue;
 
-      const ri = retainedData[uuid];
-      if ( ri && !ri._id ) ri._id = foundry.utils.randomID();
-      const item = ri ? new Item.implementation(ri) : (await fromUuid(uuid))?.clone();
-      if ( !item ) continue;
-
-      item.updateSource({
+      const itemData = retainedData[uuid] ?? (await fromUuid(uuid))?.clone().toObject();
```

Comment:
You can clean this up a bit by passing in the extra data as part of the `clone` operation:
```js
let itemData = retainedData[uuid];
if ( !itemData ) {
  const source = await fromUuid(uuid);
  itemData = source.clone({
    _id: foundry.utils.randomID(),
    "flags.dnd5e.sourceId": uuid,
    "flags.dnd5e.advancementOrigin": `${this.item.id}.${this.id}`
  }, {keepId: true}).toObject();
}
```